apps/dashboard/src/components/workflow-editor/steps/email/html-editor.tsx

```diff
@@ -102,24 +112,73 @@ export function HtmlEditor({ value, onChange }: HtmlEditorProps) {
     ];
   }, []);
 
+  const handleFormatClick = useCallback(
+    async (e: React.MouseEvent<HTMLButtonElement>) => {
+      e.stopPropagation();
+      e.preventDefault();
+
+      const formattedValue = await format(value, {
+        parser: 'liquid-html',
+        printWidth: 120,
+        tabWidth: 2,
+        useTabs: false,
+        htmlWhitespaceSensitivity: 'css',
+        plugins: [parserHtml, parserLiquid],
+      });
+
+      onChange(formattedValue);
+      saveForm();
```

Comment:
format button click handler; format the code and send the API request to save right away